python/hoomd_dlext.cc

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: MIT
+// This file is part of `hoomd-dlext`, see LICENSE.md
+
+#include "PyDLExt.h"
+#include "Sampler.h"
+
+namespace py = pybind11;
+using namespace dlext;
+
+void export_SystemView(py::module& m)
+{
+    using SystemViewSPtr = std::shared_ptr<SystemView>;
+    using PyObject = py::object;
+
+    py::class_<SystemView, SystemViewSPtr>(m, "SystemView")
+        .def(py::init<SystemDefinitionSPtr>())
+        .def("particle_data", &SystemView::particle_data)
+        .def("is_gpu_enabled", &SystemView::is_gpu_enabled)
+        .def("local_particle_number", &SystemView::local_particle_number)
+        .def("global_particle_number", &SystemView::global_particle_number)
+        .def("synchronize", &SystemView::synchronize)
+        .def("__enter__", [](SystemView& self) { self.enter(); return self; })
+        .def("__exit__", [](SystemView& self, PyObject, PyObject, PyObject) {
+            while (kPyCapsulesPool.size() > 0) {
+                invalidate(kPyCapsulesPool.back());
+                kPyCapsulesPool.pop_back();
+            }
+            self.exit();
+        })
+    ;
+}
+
+void export_PySampler(py::module m)
+{
+    using PyFunction = py::function;
+    using PySampler = Sampler<PyFunction, PyUnsafeEncapsulator>;
+    using PySamplerSPtr = std::shared_ptr<PySampler>;
+
+    py::class_<PySampler, PySamplerSPtr>(m, "DLExtSampler")
```

Comment:
suggested:
```
py::class_<PySampler, PySamplerSPtr>(m, "DLExtSampler", py::base<HalfStepHook>())
```
Otherwise, I think it doesn't know in python that we inherit from `HalfStepHook`.
And I get this error:
```TypeError: setHalfStepHook(): incompatible function arguments. The following argument types are supported:
    1. (self: hoomd._hoomd.Integrator, arg0: HalfStepHook) -> None

Invoked with: <hoomd.md._md.IntegratorTwoStep object at 0x7f0dd14120f0>, <pysages.backends.hoomd.Sampler object at 0x7f0d2c08e130>

Reply:
It should be fixed now.